### DISCUSSION TITLE
How to access webui on local network?

### DISCUSSION THREAD

**Discussion by maddes8cht** [oobabooga/text-generation-webui#1497] [Q&A] [answered]
When I open the WebUI I only see it on the localhost.
I want to share it on my local network.

I thought share=true option should solve the problem, but this generates an actual public link that makes my webapp visible from outside my network.

How can i do this?

**Comment by mastoca** [comment, marked as answer]
you should use the --listen if you want local network access.

**Comment by Pencat89** [comment]
Sorry, I'm a noob. Where do you add the --listen flag?

**Reply by jparmstr** [reply]
The easiest way: once the WebUI is running go to Interface Mode, check "listen", and click "Apply and restart the interface".

Other than that, you can edit webui.py to add the --listen flag. You can add it to the line that starts with CMD_FLAGS near the top.

**Reply by Kaszebe** [reply]
> The easiest way: once the WebUI is running go to Interface Mode, check "listen", and click "Apply and restart the interface".

This worked for me on a brand new installation of Oobabooga. Was able to access web server on same network from different computer.

**Comment by ij1738** [comment]
I'm having an issue with this where all of my IP addresses are showing up as 0.0.0.0:5000 only when --listen is added, I've already gotten --api on there and it's only when I add --listen it stops working, any help?

**Comment by verysuperkev** [comment]
In the latest versions you can edit the **CMD_FLAGS.txt** file, this allow you to add extra launch parameters.

In your case, edit the file and add --listen. Save the file and reload the start_windows,bat.

If you want to run on a different port also add --port-listen 7401 or whatever port you want to to start on.

You know it is a network accessible port as the url it tells you will be http://0.0.0.0:7401 rather than http://127.0.0.1:7401 ( this is localhost)

**Reply by VladTheDestructor** [reply]
it tells me url is http://0.0.0.0:7401/ how do i open it? when i try using link it doesn't work

**Reply by verysuperkev** [reply]
hey, you need to use the full ip or machine name i.e http://192.168.1.200:7401 - or http://MACHINE_NAME:7401

You just need to find the IP of the remote machine and use that instead of 0.0.0.0 - the zeros just mean its listening on an open network connection on your Lan.

**Reply by SvCy** [reply]
Thanks. yeah.. the IPv4 of the server (where oobabooga will run) on --listen-host.

if anyone needs it, in CMD Flags.txt at the top side under the folders, just put the line like this:
--listen --listen-host (ip4 of the server) --listen-port 8860(or your port) 
IPv4 can be found via running ipconfig or from wifi/net settings. and add --gradio-auth username:password if goes with your setup

**Comment by ij1738** [comment]
getting code code 400, message Bad request version with TLS handshake showing in the console, any idea on how to fix this or could this be an issue with my router? I've definitely port forwarded and the error only occurs when trying to connect through a mobile device

**Reply by ij1738** [reply]
Nevermind I've just realised... I've been using HTTPS instead of HTTP, sorted thank you!

**Reply by ij1738** [reply]
I am getting an error that I cannot use the same ports and the --port-listen command is unrecognized in CMD_FLAGS

**Reply by ij1738** [reply]
All good, changed networking.py to a different port

**Reply by verysuperkev** [reply]
Sorry I am a dummy, its **--listen-port** 

Its probably better to use the CMD_FLAGS file otherwise your update may get nuked when they do another commit :)

**Comment by rcorba** [comment]
I need to access Oobabooga from local LAN so I was solving the same issue ... this worked for me (dunno if it is generally the best solution): 
in .\installer_files\env\Lib\site-packages\gradio\networking.py I replaced LOCALHOST_NAME = os.getenv(“GRADIO_SERVER_NAME”, “127.0.0.1”) by LOCALHOST_NAME = os.getenv(“GRADIO_SERVER_NAME”, “0.0.0.0”) ((credit to [Kallamamran](https://github.com/Kallamamran) for his explanation in https://github.com/oobabooga/text-generation-webui/issues/1091))

I am still facing problem, that I want Oobabooga server PC go to sleep when not used. Allowing "Wake on LAN" in Bios + setting "Allow this device to wake up computer" works well, however also many other networks events are uselessly waking the server too, so it is awake all the time. Is there a change to somehow implement e.g. "magic packet" in Oobabooga?
